src/datasets/features/features.py

```diff
@@ -1755,7 +1755,13 @@ def recursive_reorder(source, target, stack=""):
                 if not isinstance(target, dict):
                     raise ValueError(f"Type mismatch: between {source} and {target}" + stack_position)
                 if sorted(source) != sorted(target):
-                    raise ValueError(f"Keys mismatch: between {source} and {target}" + stack_position)
+                    message = (
+                        f"Keys mismatch: between {source} (dataset_info.json) "
+                        f"and {target} (inferred from dataset.arrow).\n"
+                        f"{source.keys()-target.keys()} are missing from dataset.arrow "
+                        f"and {target.keys()-source.keys()} are missing from dataset_info.json" + stack_position
+                    )
```

Comment:
But if you try-except, let’s say, a `ValueError` you won’t know if it’s because of a keys mismatch or something else ("Type mismatch" although I never encountered that one).
I guess the easiest solution is just to use more generic name like "source" instead of "dataset_info.json" and "target" instead of "dataset.arrow"?

Reply:
> I guess the easiest solution is just to use more generic names like "source" instead of "dataset_info.json" and "target" instead of "dataset.arrow"?

Yes that works ! And if catch the ValueError in the `Dataset.__init__` you can re-raise to say that the issue is between the dataset_infos.json and the features from the Arrow file.

Reply:
Done! Using the same code as above it now gives:
```py
Traceback (most recent call last):
  File "/people/lerner/code/datasets/src/datasets/arrow_dataset.py", line 690, in __init__
    self.info.features = self.info.features.reorder_fields_as(inferred_features)
  File "/people/lerner/code/datasets/src/datasets/features/features.py", line 1774, in reorder_fields_as
    return Features(recursive_reorder(self, other))
  File "/people/lerner/code/datasets/src/datasets/features/features.py", line 1763, in recursive_reorder
    raise ValueError(message)
ValueError: Keys mismatch: between {'baz': Value(dtype='int64', id=None), 'bar': Value(dtype='int64', id=None)} (source) and {'foo': Value(dtype='int64', id=None), 'bar': Value(dtype='int64', id=None)} (target).
{'baz'} are missing from target and {'foo'} are missing from source

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/people/lerner/code/datasets/src/datasets/load.py", line 1853, in load_from_disk
    return Dataset.load_from_disk(dataset_path, fs, keep_in_memory=keep_in_memory)
  File "/people/lerner/code/datasets/src/datasets/arrow_dataset.py", line 1233, in load_from_disk
    return Dataset(
  File "/people/lerner/code/datasets/src/datasets/arrow_dataset.py", line 692, in __init__
    raise ValueError(
ValueError: Keys mismatch: between {'baz': Value(dtype='int64', id=None), 'bar': Value(dtype='int64', id=None)} (source) and {'foo': Value(dtype='int64', id=None), 'bar': Value(dtype='int64', id=None)} (target).
{'baz'} are missing from target and {'foo'} are missing from source
'source' means dataset_info.json and 'target' means inferred from dataset.arrow
```
pushing